### PR TITLE
[BUGFIX]: set Content-Type for non GET requests only

### PR DESCRIPTION
**Pull request by snewcomer**
- [x] ensure fetch nor ajax sets content-type header if GET request
- [x] if existing Content-Type header, don't add another one
- [x] add tests

Piggybacked off of #6169 with some tests and default content type header

close #6341 #6167

### CODE REVIEW

File: packages/adapter/addon/json-api.js [hunk 1]
```diff
@@ -143,6 +143,8 @@ import { pluralize } from 'ember-inflector';
 const JSONAPIAdapter = RESTAdapter.extend({
   defaultSerializer: '-json-api',
 
+  defaultContentType: 'application/vnd.api+json',
```
Comment (snewcomer) [inline packages/adapter/addon/json-api.js]: perhaps privatize with `_`?

File: packages/adapter/addon/rest.js [hunk 2]
```diff
@@ -1093,14 +1095,19 @@ const RESTAdapter = Adapter.extend(BuildURLMixin, {
...
 
     if (get(this, 'useFetch')) {
+      if (options.data && options.type !== 'GET') {
+        if (!options.headers['Content-Type'] && !options.headers['content-type']) {
+          options.headers['content-type'] = contentType;
+        }
+      }
       options = fetchOptions(options, this);
     } else {
+      if (options.data && options.type !== 'GET') {
```
Comment (snewcomer) [inline packages/adapter/addon/rest.js]: A GET request with a non-simple content-type will result in a preflight request.  One consequence of adding the `Content-Type` header on GET (where we used to not) is the backend server perhaps doesn't expect this and won't return the right a-c-a-o header.

File: packages/adapter/addon/json-api.js (hunk 1 above)
Comment (runspired) [reply packages/adapter/addon/json-api.js]: yes please

File: packages/adapter/addon/rest.js (hunk 2 above)
Comment (runspired) [reply packages/adapter/addon/rest.js]: Can we leave this explanation of this conditional as a comment in the code?

**Comment by brandynbennett** [comment]
@snewcomer Is this change part of any release?

**Comment by runspired** [comment]
@brandynbennett it is in beta

**Comment by brandynbennett** [comment]
Awesome thanks!

**Pull request merged**